app/services/submission_service.rb

```diff
@@ -16,7 +16,14 @@ def create_submission(submission_params, current_user)
     end
 
     def update_submission(submission, params, current_user: nil)
-      submission.assign_attributes(params)
+      if params[:user_id]
```

Comment:
Woohoo! This all seems fine to me. Would you mind adding a test to verify the behavior you've added here?

Reply:
oop yep! good thought. added in 758b30d; that feel reasonable to you?